pkg/shortnames/shortnames.go

```diff
@@ -0,0 +1,225 @@
+		normalizedShortName = strings.TrimSuffix(name, ":"+tagged.Tag())
+	}
+	return sysregistriesv2.RemoveShortNameAlias(ctx, normalizedShortName)
+}
+
+// Resolve resolves the specified name to either one or more fully-qualified
+// image references.  If the specified name is already a fully-qualified
+// reference (i.e., not a short name), it is returned as is.  In case, it's a
+// short name, it's resolved according to the ShortNameMode in the
+// SystemContext (is specified) or in the registries.conf.
+//
+// Note that returned slice may be empty if no matching alias and no
+// unqualified-search registries could be found.  Names including a digest are
+// returned as is.  Tags are stripped from the specified name.
+func Resolve(ctx *types.SystemContext, name string) ([]reference.Named, error) {
+	isShort, shortRef, err := isShortName(name)
+	if err != nil {
+		return nil, err
+	}
+	if !isShort { // no short name
+		named, isNamed := shortRef.(reference.Named)
+		if !isNamed {
+			return nil, errors.Wrapf(err, "name %q is neither a short name nor a valid named reference", name)
+		}
+		return []reference.Named{named}, nil
+	}
+
+	// Query the registry for unqualified-search registries.
+	unqualifiedSearchRegistries, err := sysregistriesv2.UnqualifiedSearchRegistries(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Detect which mode we're running in.
+	var mode types.ShortNameMode
+	if ctx != nil && ctx.ShortNameMode != nil {
+		mode = *ctx.ShortNameMode
+	} else {
+		mode, err = sysregistriesv2.GetShortNameMode(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Anonymous func to assemble (on demand!) candidates based on
+	// `unqualifiedSearchRegistries`.
+	getCandidates := func() ([]reference.Named, error) {
+		candidates := []reference.Named{}
+		for _, reg := range unqualifiedSearchRegistries {
+			named, err := reference.ParseNormalizedNamed(fmt.Sprintf("%s/%s", reg, name))
+			if err != nil {
+				return nil, errors.Wrapf(err, "error creating reference with unqalified-serach registry %q", reg)
+			}
+			candidates = append(candidates, named)
+		}
+		return candidates, nil
+	}
+
+	switch mode {
+	case types.ShortNameModeDisabled:
+		return getCandidates()
+	case types.ShortNameModePermissive, types.ShortNameModeEnforcing:
+		// Work continues after the switch.
+	default:
+		return nil, errors.Errorf("unsupported short-name mode (%v)", mode)
+	}
+
+	// Strip off the tag to normalize the short name for looking it up in
+	// the config files.
+	normalizedShortName := name
+	withTag := ""
+	if tagged, isTagged := shortRef.(reference.Tagged); isTagged {
+		withTag = tagged.Tag()
+		normalizedShortName = strings.TrimSuffix(name, ":"+withTag)
+	}
+
+	// If there's already an alias, use it.
+	isAliased, namedAlias, err := sysregistriesv2.ResolveShortNameAlias(ctx, normalizedShortName)
+	if err != nil {
+		return nil, err
+	}
+
+	if isAliased {
+		// If the input was tagged, add it to the alias.
+		if len(withTag) > 0 {
```

Comment:
`isTagged` was already a boolean. Keeping the `tagged, isTagged` variables visible here would allow dropping the `withTag` variable entirely.